src/mgr/DaemonServer.cc

```diff
@@ -410,6 +410,7 @@ bool DaemonServer::handle_report(MMgrReport *m)
     // themselves to be a daemon for some service.
     dout(4) << "rejecting report from non-daemon client " << m->daemon_name
 	    << dendl;
+    m->get_connection()->mark_down();
```

Comment:
@dillaman i agree that by forcing the client to reconnect to the mgr, MgrClient will re-send an MMgrOpen to get itself registered.

but MgrClient sends reports to mgr periodically once it's connected to the mgr. i am curious that how you managed to "pause" it to prevent it stops doing so? as the "timer" used for rescheduling `MgrClient::send_stats()` is owned by `MgrClient`, it is unlikely to be blocked by the service daemon.

Reply:
@tchaikov Would it need to pause the client from sending? The TCP connection could fail at anytime regardless (more so during test runs that purposely inject failures). The client should resend its perf stats after reconnecting (which is what I witnessed when play testing this PR after I ran into the issue originally on the dashboard).

Reply:
> The TCP connection could fail at anytime regardless (more so during test runs that purposely inject failures).

:+1: